cmds/exp/sed/sed.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// By Ahmed Kamal <email.ahmedkamal@googlemail.com>
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+)
+
+type transform struct {
+	from string
+	to   string
+}
+
+type transforms []transform
+
+func (t *transforms) String() string {
+	return fmt.Sprint(*t)
+}
+
+func (t *transforms) Set(value string) error {
+	transformDefinition := strings.Split(value, "/")
+
+	if len(transformDefinition) != 3 || transformDefinition[0] != "s" {
+		return fmt.Errorf("unable to parse transformation. This should be of the form s/old/new")
+	}
+
+	*t = append(*t, transform{from: transformDefinition[1], to: transformDefinition[2]})
+
+	return nil
+}
+
+type config struct {
+	transforms transforms
+	inplace    bool
+}
+
+var cfg = config{}
+
+func init() {
+	flag.Var(&cfg.transforms, "e", "search/replace commands (s/old/new)")
+	flag.BoolVar(&cfg.inplace, "i", false, "edit files in place")
+}
+
+func transformCopy(cfg config, readStreams []io.ReadCloser, writeStreams []io.WriteCloser) {
+	for i := range readStreams {
+		r := bufio.NewScanner(readStreams[i])
+		w := bufio.NewWriter(writeStreams[i])
+
+		for r.Scan() {
+			line := r.Text()
+			for _, transform := range cfg.transforms {
+				line = strings.ReplaceAll(line, transform.from, transform.to)
+			}
+			_, err := fmt.Fprintf(w, "%s\n", line)
+			if err != nil {
+				fmt.Printf("unable to write output: %#v", err)
+				os.Exit(1)
+			}
+		}
+
+		if err := w.Flush(); err != nil {
+			panic(err)
```

Comment:
no need to panic it's better to return error and then log.Fatalf like we do in other commands.